oval_graph/client.py

```diff
@@ -105,29 +109,54 @@ def create_dict_of_rule(self, rule_id):
             return converter.to_JsTree_dict()
         return converter.to_sigma_dict(0, 0)
 
-    def save_dict(self, dict):
-        with open(self.xml_parser.get_src(self.html_interpreter + '/data.js'), "w+") as data_file:
+    def save_dict(self, dict_, rule, date):
+        with open(os.path.join(tempfile.gettempdir(),
+                               'graph-of-' + rule +
+                               date, 'data.js'), "w+") as data_file:
             data_file.write("var data_json =" + str(json.dumps(
-                dict, sort_keys=False, indent=4) + ";"))
+                dict_, sort_keys=False, indent=4) + ";"))
+
+    def copy_interpreter(self, rule, date):
+        src = self.xml_parser.get_src(self.html_interpreter)
+        dst = os.path.join(tempfile.gettempdir(), 'graph-of-' + rule + date)
+        os.mkdir(dst)
+        for item in os.listdir(src):
+            s = os.path.join(src, item)
+            d = os.path.join(dst, item)
+            if os.path.isdir(s):
+                shutil.copytree(s, d)
+            else:
+                shutil.copy2(s, d)
 
     def prepare_data(self, rules):
         try:
+            date = str(datetime.now().strftime("_%d-%m-%Y_%H:%M:%S"))
             for rule in rules['rules']:
                 oval_tree = self.create_dict_of_rule(rule)
-                self.save_dict(oval_tree)
-                self.open_web_browser()
+                tempfile.gettempdir()
+                date = str(datetime.now().strftime("_%d-%m-%Y_%H:%M:%S"))
+                self.copy_interpreter(rule, date)
+                self.save_dict(oval_tree, rule, date)
+                self.open_web_browser(rule, date)
                 print('Rule "{}" done!'.format(rule))
+            return date
         except Exception as error:
             raise ValueError('Rule: "{}" Error: "{}"'.format(rule, error))
 
-    def open_web_browser(self):
+    def open_web_browser(self, rule, date):
         if not self.off_webbrowser:
             try:
-                webbrowser.get('firefox').open_new_tab(self.xml_parser.get_src(
-                    self.html_interpreter + '/index.html'))
+                webbrowser.get('firefox').open_new_tab(
+                    os.path.join(
```

Comment:
The path looks the same, why not to save it to a variable and use it here and in the except block?